src/components/App/App.js

```diff
@@ -0,0 +1,317 @@
+import React, { useState } from 'react';
+import { Route, Routes, useNavigate, Navigate } from 'react-router-dom';
+import { CurrentUserContext } from '../../contexts/CurrentUserContext';
+import Header from '../Header/Header';
+import Main from '../Main/Main';
+import Footer from '../Footer/Footer';
+import Login from '../Login/Login';
+import Register from '../Register/Register';
+import InfoTooltip from '../InfoTooltip/InfoTooltip';
+import MenuPopup from '../MenuPopup/MenuPopup';
+import * as NewsApi from '../../utils/NewsApi';
+import * as MainApi from '../../utils/MainApi';
+import SavedNewsHeader from '../SavedNewsHeader/SavedNewsHeader';
+import SavedNews from '../SavedNews/SavedNews';
+import ProtectedRoute from '../ProtectedRoute';
+
+function App() {
+  const navigate = useNavigate();
+  const [currentUser, setCurrentUser] = useState({});
+  const [token, setToken] = useState(localStorage.getItem('jwt'));
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [loggedInSavedNews, setLoggedInSavedNews] = useState(false);
+  const [savedNews, setSavedNews] = useState([]);
+  const [isLoginPopupOpen, setIsLoginPopupOpen] = useState(false);
+  const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
+  const [isInfoTooltipPopupOpen, setIsInfoTooltipPopupOpen] = useState(false);
+  const [isMenuPopupOpen, setIsMenuPopupOpen] = useState(false);
+  const [searchOpen, setSearchOpen] = useState(false);
+  const [preloadOpen, setPreloadOpen] = useState(false);
+  const [nothingFoundOpen, setNothingFoundOpen] = useState(false);
+  const [cards, setCards] = useState(
+    JSON.parse(localStorage.getItem('cards') || '[]')
```

Comment:
`'[]'`

The check list says: The initial value of state variables contains the correct data type.

it's a string but you need to state here an array


```js
    JSON.parse(localStorage.getItem('cards') || [])



```

Reply:
done!